### ISSUE TITLE
Define URL structure

### ISSUE THREAD

**Issue by kristinashu** [mozilla/foundation.mozilla.org#4917]
Some initial planning in [this spreadsheet](https://docs.google.com/spreadsheets/d/184z7LnHIghXDp2NIns4T8zlpBDAdaRFuPMUDX7ijHYY/edit?usp=sharing).

### Todos
- [x] Meet with designers to discuss ideal scenario
- [x] Mock-up options
- [x] Figure out how to convince everyone else
- [x] Define individual changes that are needed
- [ ] Determine which changes will need dev support

We might be blocked by approval from exec, approval is not a required step to close this ticket.

**Comment by kristinashu** [comment]
During an ad-hoc brainstorm with the designers, we came up with the idea of turning our "What We Do" main page into one long page so we can then remove the individual sub-pages for the pillars. This would allow us to still use the pillars to group content on the page but would de-emphasis their importance. It would also remove url intermediaries like `/what-we-do/rally-citizens` and link directly to things like `/what-we-do/advocacy' and `/what-we-do/tips' or `/what-we-do/research' 

@taisdesouzalessa let me know what you think the next step should be.

cc @beccaklam @sabrinang @natalieworth

**Comment by taisdesouzalessa** [comment]
I think the next steps are. 
1. to do a quick mockup of this page (main concern was how it would look and feel on mobile)
2. try to come up with 1 or 2 alternative solutions for the URL issue (mainly the "what we do" and all the other buckets such as "campaigns", "learn" and so on - the rest of the URLs seem pretty much straightforward). 

Let me know if the plan seems fine by you!

**Comment by kristinashu** [comment]
Sounds perfect!

I got a little context from Ashley about campaigns vs advocacy and added the notes to [this ticket](https://github.com/mozilla/foundation.mozilla.org/issues/3568#issuecomment-662649959). tl;dr Campaigns are a type of Advocacy work.

**Comment by taisdesouzalessa** [comment]
V1 thinking here - still need some tweaks: 
![image](https://user-images.githubusercontent.com/22157921/88347843-9dd69000-cd00-11ea-9122-343be9b85305.png)

Currently working on V2 (where all the work we do fits in one page with no tabs)

**Comment by taisdesouzalessa** [comment]
[Mockup of v2](https://redpen.io/qb00346dd88896fe9a): with all items inside one page.

Leaning more towards V2 - it seems less complicated due to current state of "buckets" definition and it will also allow us to have one less step in the our URL. Placed the v2 in Kumu so it is easier to see the paths (and it helped me to structure my thinking): https://kumu.io/taislessa/url-structure#untitled-map

Next steps: add "resources" section to Kumu (last one needed), compile thinking, present to designers and once approved, see which parts would need dev support.

**Comment by taisdesouzalessa** [comment]
Here is the final report with considerations and recommended path:
[URL Structure Report →](https://docs.google.com/document/d/1kqYux6AI1SNNu0vJI5FVFLl9gUFD_aL7CjB6SETp-Hw/edit?usp=sharing)

@kristinashu @sabrinang @natalieworth @beccaklam would you mind taking a look and let me know if you have any feedback? Feel free to comment on the doc :).

Also, if you decide to take a look at the maps in Kumu, you can select a page and click on "focus" to see all connections from and to that page. It clears the view a bit and helps understanding all the details. 
<img width="576" alt="image" src="https://user-images.githubusercontent.com/22157921/88729454-ce477100-d0e8-11ea-8dcf-84c1d3191570.png">

**Comment by taisdesouzalessa** [comment]
Maybe some prompts will help since I think it may be overwhelming to look at those diagrams at a glance:

- do you prefer V1 or V2? And let me know why, please
- Do the structures/names of URL make sense to you?
- Do you see any issues there? Isolated hubs? We have a "resources" (currently "doc") that is isolated from the pages (there is no link to that hub anywhere) - but this was intentional, I believe, because we don't want the general audience accessing those pages, right?
- What do you think of having a section called "Archive" at the bottom of "what we do" page - very subtle, so people can access old projects from there?
- Anything else I missed?

**Comment by kristinashu** [comment]
Wonderful, thank you @taisdesouzalessa for the report! There's some interesting info in there. It helped be accept the use of us having many many primary level pages, which I have always struggled with.

For your prompts:

### Do you prefer V1 or V2? And let me know why, please

From our talks I think we were in consistence that we prefer V2 because issues we had issues with V1 e.g. 
- it forced our work into buckets when in reality there is a lot of overlap between them
- we couldn't align with execs on the bucket names so de-emphasizing them seemed like a nice solution
- some of the page content seemed redundant

Our concern was that V2 might be too long but your comps showed that it wasn't! 

### Do the structures/names of URL make sense to you?

I think so but it's hard for me to see all the exact URL names within the doc. Do you mind updating or commenting in the [spreadsheet](https://docs.google.com/spreadsheets/d/184z7LnHIghXDp2NIns4T8zlpBDAdaRFuPMUDX7ijHYY/edit#gid=0) on what the structures/names are?

### Do you see any issues there? Isolated hubs? We have a "resources" (currently "doc") that is isolated from the pages (there is no link to that hub anywhere) - but this was intentional, I believe, because we don't want the general audience accessing those pages, right?

"docs" is very different than all the other pages. This is for internal staff or contractor use and we definitely should not link to it from anywhere on the site.

### What do you think of having a section called "Archive" at the bottom of "what we do" page - very subtle, so people can access old projects from there?

I think "archive" should remind buried and should not be surfaced on /what-we-do. I don't see the value in new audiences seen all that. The info on those pages isn't very useful or well maintained and in some ways it even highlights our disjointed past.

### Anything else I missed?

I'm conflicted on where work like a Misinformation or Elections hub should live? They seem to be a mix of "learn" and "campaign". I don't think it's something we would do year over year so probably shouldn't warrant being a primary page, but should it? Not sure you can answer that tho.

**Comment by taisdesouzalessa** [comment]
Added the proposed structure to the [spreasdheet (second tab)](https://docs.google.com/spreadsheets/d/184z7LnHIghXDp2NIns4T8zlpBDAdaRFuPMUDX7ijHYY/edit?usp=sharing)

I find the visual map helps convince people and make them understand that URL depth is not the same as IA depth. So checked the 3rd box (let me know if I misunderstood what you meant by that, @kristinashu . The individual changes that are needed are highlighted on the spreadsheet.

The only missing thing is "Determine which changes will need dev support" - as per our discussion today I may need to do a map of redirects and may need to involve engineers to better understand how this works. I'll open a new ticket for that tomorrow as I don't think it will be done this sprint.

**Comment by taisdesouzalessa** [comment]
Couldn't do the last checkmark - opened a new ticket for it to be done next sprint: #4999
